examples/convert_mri_and_trans.py

```diff
@@ -24,19 +24,145 @@
+output_path = op.abspath(op.join(data_path, '..', 'MNE-sample-data-bids'))
+if op.exists(output_path):
+    sh.rmtree(output_path)
+raw = mne.io.read_raw_fif(raw_fname)
+sub = '01'
+ses = '01'
+task = 'audiovisual'
+run = '01'
+bids_basename = make_bids_basename(subject=sub, session=ses, task=task,
+                                   run=run)
+write_raw_bids(raw, bids_basename, output_path, events_data=events_data,
+               event_id=event_id, overwrite=True)
+
+# Print the directory tree
+print_dir_tree(output_path)
+
+###############################################################################
+# Now let's assume that we have also collected some T1 weighted MRI data for
+# our subject. And furthermore, that we have already aligned our coordinate
+# frames (as described HERE) and obtained a transformation matrix
+# :code:`trans`.
+
+# Get the path to our MRI scan and covert it to NIfTI format
+# For that, we use nibabel, reading in the compressed mgh file, and saving it
+# as a compressed NIfTI file
+t1_mgh_fname = op.join(data_path, 'subjects', 'sample', 'mri', 'T1.mgz')
+t1_mgh = nib.load(t1_mgh_fname)
+t1_nii_fname = op.join(data_path, 'subjects', 'sample', 'mri', 'T1.nii.gz')
+nib.save(t1_mgh, t1_nii_fname)
+
+# Load the transformation matrix and show what it looks like
+trans_fname = op.join(data_path, 'MEG', 'sample',
+                      'sample_audvis_raw-trans.fif')
+trans = mne.read_trans(trans_fname)
+print(trans)
+
+###############################################################################
+# We can save the MRI to our existing BIDS directory and at the same time
+# create a JSON sidecar file that contains metadata, we will later use to
+# retrieve our transformation matrix :code:`trans`.
+
+# We use the write_anat function
+write_anat(bids_root=output_path,  # point to the BIDS dir we wrote earlier
+           subject=sub,
+           t1w=t1_nii_fname,  # path to the MRI scan in NIfTI format
+           session=ses,
+           raw=raw,  # the raw MEG data file connected to the MRI
+           trans=trans,  # our transformation matrix
+           verbose=True  # this will print out the sidecar file
+           )
+
+# Clean up the NIfTI file we wrote earlier to convert from mgh to NIfTI
+os.remove(t1_nii_fname)
+
+# Let's have another look at our BIDS directory
+print_dir_tree(output_path)
+
+###############################################################################
+# Our BIDS dataset is now ready to be shared. We can easily reproduce our
+# transformation matrix using ``MNE-BIDS`` and the BIDS dataset.
+
+bids_fname = bids_basename + '_meg.fif'
+
+# reproduce our trans
+_trans = get_head_mri_trans(bids_fname=bids_fname,  # name of the MEG file
+                            bids_root=output_path  # root of our BIDS directory
+                            )
+
+# Show that they are the same
+np.testing.assert_almost_equal(trans['trans'],
+                               _trans['trans'])
```

Comment:
this is for a test not an example

Reply:
haha, we had this discussion before (a year ago or so) ... I still feel that this can serve as a "nice demo to proof that the arrays are really the same" ... so that users don't have to trust.

I'll remove it, because it's two mentors against one tiny student :)